rust/geoparquet/src/writer/metadata.rs

```diff
@@ -171,13 +171,20 @@ impl GeoParquetMetadataBuilder {
     pub fn try_new(schema: &Schema, options: &GeoParquetWriterOptions) -> GeoArrowResult<Self> {
         let mut columns = HashMap::new();
 
+        let mut options_primary_column_seen = !options.primary_column.is_some();
         for (col_idx, field) in schema.fields().iter().enumerate() {
             if let Some(ext_name) = field.metadata().get(EXTENSION_TYPE_NAME_KEY) {
                 if !ext_name.starts_with("geoarrow") {
                     continue;
                 }
 
                 let column_name = schema.field(col_idx).name().clone();
+                options_primary_column_seen = !options_primary_column_seen
+                    && options
+                        .primary_column
+                        .as_deref()
+                        .map(|c| c == column_name)
+                        .unwrap_or_default();
```

Comment:
It's a bit opaque to read, but lets us check as we iterate through the schema fields?